__init__.py

```diff
@@ -396,6 +374,47 @@ def handle_latest_news(self, message=None, feed=None):
             self.log.info("Traceback: {}".format(traceback.format_exc()))
             self.speak_dialog("could.not.start.the.news.feed")
 
+    def _play_station(self, station_title, station_url, station_image):
+        """Play the given station using the most appropriate service."""
+        # TODO convert all station references to named tuples.
+        self.log.info(f'Playing News feed: {station_title}')
+        media_url = self.get_media_url(station_url)
+        self.log.info(f'News media url: {media_url}')
+        mime = find_mime(media_url)
+        # Ensure announcement of station has finished before playing
+        wait_while_speaking()
+        # If backend cannot handle https, download the file and provide a local stream.
+        if media_url[:8] == 'https://' and not self.is_https_supported():
+            self.download_media_file(media_url)
+            self.CPS_play((f"file://{self.STREAM}", mime))
+        else:
+            self.CPS_play((media_url, mime))
+        self.CPS_send_status(
+            image=station_image or image_path('generic.png'),
+            artist=station_title
+        )
+        self.now_playing = station_title
+
+    def is_https_supported(self):
+        """Check if any available audioservice backend supports https"""
+        for name, service in self.audioservice.available_backends().items():
+            if 'https' in service['supported_uris']:
+                return True
+        return False
+
+    def download_media_file(self, url):
+        """Download a media file and return path to the stream"""
+        # (Re)create Fifo
+        if os.path.exists(self.STREAM):
+            os.remove(self.STREAM)
+        os.mkfifo(self.STREAM)
+        self.log.debug('Running curl {}'.format(url))
+        args = ['curl', '-L', quote(url, safe=":/"), '-o', self.STREAM]
+        self.curl = subprocess.Popen(args)
+        # Check if downloaded file is actually an error page
+        if contains_html(self.STREAM):
+            raise Exception('Could not fetch valid audio file.')
```

Comment:
We should never raise a base exception. If one of the built-in exception classes don't work, define a new one.  this promotes having less bare except clauses in `try` blocks